demo/src/main/java/foundation/algorand/demo/AnswerActivity.kt

```diff
@@ -478,70 +490,38 @@ class AnswerActivity : AppCompatActivity() {
         try {
-            val message = JSONObject(msgStr)
-            if (message.get("type") == "transaction") {
-                lifecycleScope.launch {
-                    // Decode the Transaction
-                    val txn = decodeUnsignedTransaction(message.get("txn").toString())
-                    // Display a biometric prompt with some transaction details
-                    val biometricResult = biometrics(txn!!)
-                    if (biometricResult !== null) {
-                        val bytes = txn.bytesToSign()
-                        val signatureBytes = KeyPairs.rawSignBytes(bytes, keyPair.private)
-                        val sig = Base64.encodeBase64URLSafeString(signatureBytes)
-                        val responseObj = JSONObject()
-                        responseObj.put("sig", sig)
-                        responseObj.put("txId", txn.txID())
-                        responseObj.put("type", "transaction-signature")
-                        signalService!!.send(responseObj.toString())
-                        signalService!!.notify(
-                            notifications.createNotificationBuilder(this@AnswerActivity)
-                                .setOnlyAlertOnce(true)
-                                .setContentIntent(signalService!!.createPendingIntent(AnswerActivity::class.java))
-                            ,
-                            NotificationViewModel.SERVICE_NOTIFICATION_ID
-                        )
-                        // Is Notification Intent(not Deep Link)
-                        if (intent?.data == null && signalService!!.isDeepLink) {
-                            intent?.let { deepLinkIntent ->
-                                if (deepLinkIntent.getStringExtra("msg") !== null) {
-                                    signalService!!.lastKnownReferer?.let { referer ->
-                                        this@AnswerActivity.finish()
-                                        val browserIntent = packageManager.getLaunchIntentForPackage(
-                                            referer.replace(
-                                                "android-app://",
-                                                ""
-                                            )
-                                        )
-                                        browserIntent?.let { openBrowser ->
-                                            openBrowser.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
-                                            startActivity(browserIntent)
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                    }
+            // TODO: Refactor to ByteArray and allow streaming of the Buffer
+            val resultMessage = provider.handleRequestMessage(Message(Base64.UrlSafe.decode(msgStr), EncoderType.CBOR), keyPair)
```

Comment:
`handleRequestMessage` should probably be from an agnostic interface that any AVM / Identity / other context can implement so that. Can we do that? Should be straight forward; maybe the interface can live in `liquid` code package

Reply:
Should be that way now, it does accepts a generic Message which is only the bytes and the encoding (not specifying ARC27 schema).

I named the method handleRequestMessage since it only handles `RequestMessages` from ARC27 (the most common use case for the Wallet and the AVMProvider). I could refactor it to handleMessage and have it support both?